lib/alephant/logger/level.rb

```diff
@@ -0,0 +1,38 @@
+module Alephant
+  module Logger
+    LEVELS = %i(debug info warn error).freeze
+
+    class Level
+      def initialize(desired, defined)
+        @desired = desired
+        @defined = defined
+      end
+
+      def log?
+        desired_level <= defined_level
+      end
+
+      private
+
+      attr_reader :desired, :defined
+
+      def desired_level
+        case desired
+        when Symbol then log_level(desired) || 0
+        when Integer then desired >= 0 ? desired : 0
```

Comment:
Do we need this logic? If greater than 0, then return, else return 0. Why not just return desired, or should we check if desired is in the range of levels?

Reply:
@samfrench This is to cover the case where desired might be less than 0 e.g. `-1`. IMO, I think it's valid to setup this validation.

Further, if `desired` is greater than the size of the levels, `#log?` will return false since the check is ` desired_level <= defined_level`.

Reply:
Aaah, I see what you mean. Good shout Sam! 🙂